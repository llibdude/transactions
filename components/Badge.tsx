import cn from "classname";

type Props = {
  variant: "red" | "green";
  children: React.ReactNode;
};

const Badge: React.FC<Props> = ({ children, variant }) => {
  const classes = cn(
    "inline-flex items-center px-1.5 py-sm border rounded-full text-sm font-medium",
    {
      "border-red-500 bg-red-50 text-red-600": variant === "red",
      "border-green-500 bg-green-50 text-green-600": variant === "green",
    }
  );

  return <div className={classes}>{children}</div>;
};

export default Badge;
