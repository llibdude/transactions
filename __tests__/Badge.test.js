import { render } from "@testing-library/react";
import Badge from "../components/Badge";

// Simple UI components are a good use case for snapshot testing.
// I personally find when you do snapshot testing with more volatile code it can distract from the teams efficiency.
it("renders red badge", () => {
  const { container } = render(<Badge variant="red">Test content</Badge>);
  expect(container).toMatchSnapshot();
});

it("renders green badge", () => {
  const { container } = render(<Badge variant="green">Test content</Badge>);
  expect(container).toMatchSnapshot();
});
