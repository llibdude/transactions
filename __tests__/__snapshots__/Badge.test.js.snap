// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders green badge 1`] = `
<div>
  <div
    class="inline-flex items-center px-1.5 py-sm border rounded-full text-sm font-medium border-green-500 bg-green-50 text-green-600"
  >
    Test content
  </div>
</div>
`;

exports[`renders red badge 1`] = `
<div>
  <div
    class="inline-flex items-center px-1.5 py-sm border rounded-full text-sm font-medium border-red-500 bg-red-50 text-red-600"
  >
    Test content
  </div>
</div>
`;
